hw/bus/drivers/i2c_da1469x/src/i2c_da1469x.c

```diff
@@ -0,0 +1,677 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+#include <assert.h>
+#include <DA1469xAB.h>
+#include <mcu/da1469x_pd.h>
+#include "defs/error.h"
+#include "hal/hal_gpio.h"
+#include "bus/bus.h"
+#include "bus/bus_debug.h"
+#include "bus/bus_driver.h"
+#include "bus/drivers/i2c_common.h"
+#include "mcu/mcu.h"
+
+#if MYNEWT_VAL(I2C_DA1469X_BUS_DRIVER)
+#include "mcu/da1469x_dma.h"
+#endif
+
+#if MYNEWT_VAL(I2C_DA1469X_STAT)
+#include "stats/stats.h"
+#endif
+
+#if MYNEWT_VAL(I2C_DA1469X_STAT)
+STATS_SECT_START(i2c_da1469x_stats_section)
+    STATS_SECT_ENTRY(dma_read_bytes)
+    STATS_SECT_ENTRY(dma_written_bytes)
+    STATS_SECT_ENTRY(i2c_errors)
+STATS_SECT_END
+
+STATS_NAME_START(i2c_da1469x_stats_section)
+    STATS_NAME(i2c_da1469x_stats_section, dma_read_bytes)
+    STATS_NAME(i2c_da1469x_stats_section, dma_written_bytes)
+    STATS_NAME(i2c_da1469x_stats_section, i2c_errors)
+STATS_NAME_END(i2c_da1469x_stats_section)
+#endif
+#define I2C_COUNT 2
+
+static void i2c_da1469x_i2c0_isr(void);
+static void i2c_da1469x_i2c1_isr(void);
+
+struct da1469x_i2c_hw {
+    I2C_Type *regs;
+    uint8_t scl_func;
+    uint8_t sda_func;
+    IRQn_Type irqn;
+    /* DMA peripheral index */
+    uint8_t dma_periph;
+    /* Bit number for clock selection in CLK_COM_REG */
+    uint8_t clk_src_bit;
+    /* Bit number for clock enable in CLK_COM_REG */
+    uint8_t clk_ena_bit;
+    void (* isr)(void);
+};
+
+static const struct da1469x_i2c_hw da1469x_i2c[2] = {
```

Comment:
Should syscfg knobs I2C_0 and I2C_1 be inspected before defining these? Specifically I could see users not defining pins for I2C bus which is not enabled.